### ISSUE TITLE
Specific OData function parameter value causing 404 response (parameter value of -119.4878 causes 404, values of -119.4877, -119.4879, -119.48781, -119.487 don't)

### ISSUE THREAD

**Issue by JShawnLooney** [OData/odata.net#1403]
When passing a specific OData function parameter double value, it is causing 404 response before validating route.  Maybe a bug in the URL parser.

For example, passing a value of -119.4878 (the longitude of Benton City, WA) to the function causes 404, but values of -119.4877, -119.4879, -119.48781, or-119.487 don't.  

### Reproduce steps

See attached solution zip.  Build and execute ExternalLGData.Application.  Execute the following API calls to reproduce:
`https://localhost:5001/api/PropertyTaxBoundaries(2017)/Boundaries/Contains(Latitude=46.2632,Longitude=-119.4878)`
[this returns a 404 error - not found]

[ExternalLGData.Sample.zip](https://github.com/OData/odata.net/files/2858442/ExternalLGData.Sample.zip)

### Expected result
Strangely, if I add 0.00001 to the longitude, it works.  Or change the last digit from 8 to 7 or 9. 
`https://localhost:5001/api/PropertyTaxBoundaries(2017)/Boundaries/Contains(Latitude=46.2632,Longitude=-119.48781)`
[this returns correct results; notice I just added 0.00001 to the longitude]
`https://localhost:5001/api/PropertyTaxBoundaries(2017)/Boundaries/Contains(Latitude=46.2632,Longitude=-119.4877)`
[this returns correct results; notice I just changed the 8 to a 7 in the longitude]
`https://localhost:5001/api/PropertyTaxBoundaries(2017)/Boundaries/Contains(Latitude=46.2632,Longitude=-119.4879)`
[this returns correct results; notice I just changed the 8 to a 9 in the longitude]

### Additional detail
.NET Core 2.2
OData 7.1.0
EF Core 2.2.2
Same results running from VS2017 or deployed to IIS (Self Contained exe or dll) 
I've commented out all middleware (CORS, MemoryCache)...same results.

**Comment by JShawnLooney** [comment]
Related to old similar bug? #3

**Comment by xuzhg** [comment]
@JShawnLooney See some discussion in https://github.com/dotnet/coreclr/issues/22587

**Comment by JShawnLooney** [comment]
@xuzhg Thanks for tracking down the cause.  It sounds like this won't be an issue in .NET Core 3.0 then, but  what if any options do we have under the current .NET Core 2.2?

**Comment by xuzhg** [comment]
I think change the type of parameter as "decimal" can work. 
From the latest comments in dotnet/coreclr#22587, it seems our codes [here](https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs#L1177-L1198) has problem in different platform and different version.

**Comment by xuzhg** [comment]
@JShawnLooney Or, you can use the 'd' suffix after the value. for example: 

`.../Contains(Latitude=46.2632,Longitude=-119.4878d)`

It works but in the controller method, you will get a inaccurate value. maybe: -119.48779999999999

**Comment by JShawnLooney** [comment]
@xuzhg Yep, but we've got a number of consumers already who I'd prefer not to impact, so I'm changing my function parameter types to take decimals.  

FYI, I can confirm that switching the function parameter to float does not work and still gets hung up in the parsing and I get a 404 for that same value.  

Thank you again for tracking down the cause of this issue.  This gives me what I need for my workaround, but I will leave it to you to keep this issue open if you need it to track OData code changes.

**Comment by xuzhg** [comment]
@JShawnLooney Thanks, let us keep this issue open to track on.
